sam.c

```diff
@@ -1304,17 +1304,16 @@ int bam_aux_update_str(bam1_t *b, const char tag[2], int len, const char *data)
     bam_aux_del(b,s);
```

Comment:
Perhaps not; if anything the whole line could be replaced with `assert`.  This is programmer error (either wrong tag used or failure to check tag is type `Z`) rather than runtime failure or user error, so should fail immediately and gracelessly.